openhcl/virt_mshv_vtl/src/processor/tdx/mod.rs

```diff
@@ -1349,7 +1349,14 @@ impl UhProcessor<'_, TdxBacked> {
                     })
                     .msr_read(msr)
                     .or_else_if_unknown(|| self.read_msr(msr, intercepted_vtl))
-                    .or_else_if_unknown(|| self.read_msr_cvm(msr, intercepted_vtl));
+                    .or_else_if_unknown(|| self.read_msr_cvm(msr, intercepted_vtl))
+                    .or_else_if_unknown(|| match msr {
+                        hvdef::HV_X64_MSR_GUEST_IDLE => {
+                            self.backing.lapic.halted = true;
```

Comment:
Don't we need to treat this slightly differently than `hlt`? I thought the idea was that if we're in the idle state, we don't take IF into account when deciding whether to unhalt.

Reply:
Ah, you are right. As the first iteration was written, I believe it worked because we cleared halt whenever rvi was non-zero when offloaded, but it would not have worked when non-offloaded (as it would've respected IF, at least on TDX). Hopefully the newer version is more correct